sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -266,27 +292,135 @@ private boolean validateSortDocument(List<Map<String, String>> sortDocument) {
                                  List<Map<String, String>> sortDocument,
                                  Map<String, Object> indexes,
                                  SQLDatabase db) {
-        // TODO - sort logic...
-        return null;
+        boolean smallResultSet = (docIdSet.size() < SMALL_RESULT_SET_SIZE_THRESHOLD);
+        SqlParts orderBy = sqlToSortIds(docIdSet, sortDocument, indexes);
+        List<String> sortedIds = new ArrayList<String>();
```

Comment:
Changed sortedIds assignment inside `else` block to mirror objc implementation.  https://github.com/cloudant/sync-android/commit/6179fea1b1139126f9e958e9eaf59fdea174709a#diff-80ec4d1d12847da53428d6c14a88dbb7R331